File: development/palindrome_word/palindrome_word.py
def is_palindrome(word):
    left = 0
    right = len(word) -1

    while left < right:
        if word[left] != word[right]:
            return False
        
        left += 1
        right -= 1

    return True

word1 = "9875789"
word2 = "deep learning"
print("the word: %s is palindrome %t", word1, is_palindrome(word1))
print("the word: %s is palindrome %t", word2, is_palindrome(word2))

File: development/palindrome_word/palindrome_word.go
package main

import "fmt"

func isPalindrome(word string) bool {
	left := 0
	right := len(word) - 1

	for left < right {
		if word[left] != word[right] {
			return false
		}

		left++
		right--
	}

	return true
}

func main() {
	word1 := "123404321"
	word2 := "machine learning"
	fmt.Println("the word: %s is a palindrome: %t", word1, isPalindrome(word1))
	fmt.Println("the word: %s is a palindrome: %t", word2, isPalindrome(word2))
}
